### PR TITLE
[Functions] ToS allow list

### PR DESCRIPTION
**Pull request by justinkaseman**
### Changes
- Use ReadAccessController interface within ToS Allowlist
- add method for get all

### CODE REVIEW

File: contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol
Comment (bolekk) [inline contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol:128]: why two methods?

File: contracts/src/v0.8/functions/dev/1_0_0/accessControl/interfaces/ITermsOfServiceAllowList.sol
Comment (bolekk) [inline contracts/src/v0.8/functions/dev/1_0_0/accessControl/interfaces/ITermsOfServiceAllowList.sol:46]: What do you think about adding a paginated version? There would be a getAllowlistSize() and then getAllowlistSlice(start, end)?

Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/accessControl/interfaces/ITermsOfServiceAllowList.sol:46]: I actually don't think we should worry about the allow list getting too large.
As a view function called off-chain we shouldn't have any problems.
The risk is calling it on-chain and running out of gas, but that's not our use case.

File: contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol:128]: `hasAccess` has that data field, which is fairly annoying to call.
Easier to allow the overload.

**Review by bolekk — DISMISSED** [review]

File: contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol
Comment (se3000) [reply contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol:128]: I'm with Bolek, I think just one method where you pass empty bytes is better. I'm pretty sure that adding another public method will take up more bytecode space.

**Review by bolekk — DISMISSED** [review]

File: contracts/src/v0.8/functions/dev/1_0_0/accessControl/interfaces/ITermsOfServiceAllowList.sol
Comment (KuphJr) [reply contracts/src/v0.8/functions/dev/1_0_0/accessControl/interfaces/ITermsOfServiceAllowList.sol:46]: View functions still have a "virtual" gas limit equal to the block gas limit.  An SLOAD takes 800 gas, plus whatever the additional gas cost is to construct the array, so we would exceed Ethereum's 30mil block gas limit at around 30,000 addresses or so.

File: contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol]: Changed

**Review by bolekk — APPROVED** [review]

File: contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol
Comment (KuphJr) [inline contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol:114]: This allowlist contract is upgradeable, so I am not going to block the PR because of this, but we do need to add a start & end once our user count grows past a few thousand.

File: contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol
Comment (KuphJr) [reply contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol]: How much more gas does `haveAccess` require if we need to pass this unused data argument?  Probably not much, but it the interest of gas golf it might be a good optimization.

**Review by KuphJr — APPROVED** [review]
Left some comments you should read before merging, but I am approving.

**Pull request merged**
